gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/controller/OrganizationController.java

```diff
@@ -28,53 +35,67 @@ public class OrganizationController {
 
     private OrganizationService organizationService;
     private JwtEncoder jwtEncoder;
+    private OrganizationConverter organizationConverter;
+    private UserOrganizationService userOrganizationService;
 
     @Autowired
     public OrganizationController(OrganizationService organizationService,
-                                  JwtEncoder jwtEncoder){
+                                  JwtEncoder jwtEncoder,
+                                  OrganizationConverter organizationConverter,
+                                  UserOrganizationService userOrganizationService) {
         this.organizationService = organizationService;
         this.jwtEncoder = jwtEncoder;
+        this.organizationConverter = organizationConverter;
+        this.userOrganizationService = userOrganizationService;
     }
 
 
     @PreAuthorize("@securityUtils.hasAuthorityToRequestedOrgId('OP_READ_DOCUMENT')")
     @GetMapping("/organizations")
-    public Page<Organization> getAllOrganizations(@Valid OrganizationCriteria critetia, Pageable pageable) throws Exception{
+    public Page<Organization> getAllOrganizations(@Valid OrganizationCriteria critetia, Pageable pageable) throws Exception {
 
         //run code to get the organization from database
         return organizationService.getAllOrganizations(critetia, pageable);
     }
 
     @GetMapping("/organizations/{id}")
-    public Organization getOrganizationById(@PathVariable UUID id, Authentication authentication) throws Exception{
+    public Organization getOrganizationById(@PathVariable UUID id, Authentication authentication) throws Exception {
 
         //run code to get the organization from the database
         Organization organization = organizationService.getById(id);
         return organization;
     }
 
-    @PostMapping(value="/organizations", consumes = {"application/json"})
+    @PostMapping(value = "/organizations", consumes = {"application/json"})
     @ResponseStatus(value = HttpStatus.CREATED)
-    public Organization createOrganization(@RequestBody OrganizationDTO organizationDTO) throws Exception{
-        Organization organization = new Organization();
-        Instant now= Instant.now();
-        organization.setName(organizationDTO.getName());
-        organization.setDisplayName(organizationDTO.getDisplayName());
-        organization.setAddress(organizationDTO.getAddress());
-        organization.setPhone(organizationDTO.getPhone());
+    public Organization createOrganization(@RequestBody OrganizationDTO organizationDTO) throws Exception {
+        Organization organization = organizationConverter.toEntity(organizationDTO);
+        Instant now = Instant.now();
         organization.setCreatedAt(now);
         organization.setUpdatedAt(now);
         organization = organizationService.createOrganization(organization);
+
+        userOrganizationService.setAdminRoleForOrganizationsOwner(organization);
```

Comment:
I think, this is also **business logic**, so needs to be done in the method `organizationService.createOrganization`, not in the Controller